sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/ZetaSqlUtils.java

```diff
@@ -196,7 +196,55 @@ private static Value beamLogicalObjectToZetaSqlValue(Object object, String ident
     }
   }
 
-  // Value conversion: ZetaSQL => Beam
+  // Type conversion: ZetaSQL => Beam
+  public static FieldType zetaSqlTypeToBeamFieldType(Type type) {
```

Comment:
Out of curiosity: so the ZetaSQL type does not carry nullability? (otherwise seems the nullability of the type has lost in this conversion).

Reply:
Yes. All ZetaSQL types are nullable. i.e. You can't define a ZetaSQL type that disallow null values.